torch/nn/parallel/distributed.py

```diff
@@ -125,6 +128,16 @@ def _dump_DDP_relevant_env_vars():
     print(formatted_output)
 
 
+class _BufferCommHookLocation(Enum):
+    PRE_FORWARD = auto()
+    POST_FORWARD = auto()
+
+@dataclass
+class _BufferCommHook:
+    buffer_comm_hook: Callable
```

Comment:
nit: document how the signature of a buffer_comm_hook looks like: 

def buffer_comm_hook(state, self.name_buffers)

Reply:
I added the documentation to the function in the PR at the top of this stack: https://github.com/pytorch/pytorch/pull/64515/files, will just leave it there if you're okay so I don't have to deal with merge conflicts.